src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopCryptographyManager.cs

```diff
@@ -85,10 +92,21 @@ public byte[] SignWithCertificate(string message, X509Certificate2 certificate)
                 return rsa.SignData(messageBytes, sha);
             }
 #else
-            using (var key = certificate.GetRSAPrivateKey())
+            if (!CertificateToRsaMap.TryGetValue(certificate.Thumbprint, out RSA rsa))
             {
-                return key.SignData(messageBytes, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                if (CertificateToRsaMap.Count == maximumCertificateToRsaMap)
+                {
+                    ClearCertificateToRsaMap clearCertificateToRsaMap = new ClearCertificateToRsaMap(CertificateToRsaMap, clearMapLock);
+                    (new Thread(new ThreadStart(clearCertificateToRsaMap.Clear))).Start();
```

Comment:
using a Task might be preferable here

Reply:
This code pulled from a stale branch, disposing the RSA objects is fraught with issues.
Instead, just clear the dictionary.
If the number is reasonable, say 100 or 200, the memory pressure should be minimal.